packages/flutter_tools/lib/src/flx.dart

```diff
@@ -36,7 +36,7 @@ Future<int> createSnapshot({
   assert(mainPath != null);
   assert(snapshotPath != null);
   assert(packages != null);
-  final String snapshotterPath = artifacts.getArtifactPath(Artifact.genSnapshot);
+  final String snapshotterPath = artifacts.getArtifactPath(Artifact.genSnapshot, null, BuildMode.debug);
```

Comment:
Got it. I think this will break getting the artifact in the non-local engine mode because that code currently does not support the combination of `mode != null` and `targetPlatform == null`.